### PR TITLE
testing: no more inout parameters in bottom-up test

### PR DESCRIPTION
**Pull request by superlopuh**
This bit will make it easier to raise to memref and scf, the last non-riscv bits.

I visually compared these to the new kernels' assembly in the epxeriments repo, and they look pretty much the same.

### CODE REVIEW

File: tests/filecheck/projects/riscv-backend-paper/bottom_up.mlir [hunk 1]
```diff
@@ -500,12 +511,16 @@ func.func public @pooling_nchw_max_d1_s2_3x3(
...
+// CHECK-NEXT:  .text
 // CHECK-NEXT:  .globl pooling_nchw_max_d1_s2_3x3
 // CHECK-NEXT:  .p2align 2
 // CHECK-NEXT:  pooling_nchw_max_d1_s2_3x3:
-// CHECK-NEXT:      mv t2, a0
-// CHECK-NEXT:      mv t0, a1
+// CHECK-NEXT:      mv t3, a0
+// CHECK-NEXT:      mv t1, a1
+// CHECK-NEXT:      li t4, -1060927488
+// CHECK-NEXT:      sw t4, -4(sp)
```
Comment (superlopuh) [inline tests/filecheck/projects/riscv-backend-paper/bottom_up.mlir:521]: this is the only bit that I haven't tested, but it should be ok

**Comment by AntonLydike** [comment]
> I visually compared these to the new kernels' assembly in the epxeriments repo, and they look pretty much the same.

lol :smile:

File: tests/filecheck/projects/riscv-backend-paper/bottom_up.mlir (hunk 1 above)
Comment (AntonLydike) [reply tests/filecheck/projects/riscv-backend-paper/bottom_up.mlir:521]: Should be fine, we don't really respect the stack much anyway right now :smile:

**Review by AntonLydike — APPROVED** [review]
Yep! This looks good to me! :+1:

**Pull request merged**